File: portfolio-srv/src/Pi.PortfolioService/Pi.PortfolioService/Models/TradingAccount.cs
namespace Pi.PortfolioService.Models;

public enum Product
{
    Unknown,
    Cash,
    CashBalance,
    CreditBalanceSbl,
    Crypto,
    Derivatives,
    GlobalEquities,
    Funds,
    Bond,
    CashSbl,
    CashBalanceSbl,
    CreditBalance,
    StructureNoteOnShore,
    Dr,
    LiveX,
    BorrowCash,
    BorrowCashBalance
}

public class TradingAccount
{
    public TradingAccount(Guid userId, string customerCode, string tradingAccountNo, Product product)
    {
        CustomerCode = customerCode;
        TradingAccountNo = tradingAccountNo;
        UserId = userId;
        Product = product;
    }

    public string CustomerCode { get; }
    public string TradingAccountNo { get; }
    public Guid UserId { get; }
    public Product Product { get; }
}

File: bond-srv/internal/domain/account/trading_account.go
package domain

import "strings"

type TradingAccount struct {
	CustomerCode     string
	AccountNo        string //custCodesuffix
	TradingAccountNo string //custCode-suffix
	Product          Product
}

type Product int

const (
	Cash        Product = iota
	CashBalance Product = iota
	Bond        Product = iota
)

func NewTradingAccount(customerCode, tradingAccountNo string, product Product) *TradingAccount {
	return &TradingAccount{
		CustomerCode:     customerCode,
		TradingAccountNo: tradingAccountNo,
		AccountNo:        strings.Replace(tradingAccountNo, "-", "", 1),
		Product:          product,
	}
}

func (t *TradingAccount) IsBondAccount() bool {
	return t.Product == Bond
}
